packages/contentstack-audit/src/modules/entries.ts

```diff
@@ -109,11 +110,36 @@ export default class Entries {
             if (!this.missingSelectFeild[this.currentUid]) {
               this.missingSelectFeild[this.currentUid] = [];
             }
+
+            if (!this.missingMandatoryFields[this.currentUid]) {
+              this.missingMandatoryFields[this.currentUid] = [];
+            }
             if (this.fix) {
               this.removeMissingKeysOnEntry(ctSchema.schema as ContentTypeSchemaType[], this.entries[entryUid]);
             }
 
             this.lookForReference([{ locale: code, uid, name: title }], ctSchema, this.entries[entryUid]);
+            
+            if(this.missingMandatoryFields[uid] && entry.publish_details.length){
+              const isPublished = entry.publish_details.length > 0;
+              const fixStatus = this.fix ? 'Fixed' : '';
+  
+              this.missingMandatoryFields[uid]?.forEach((field: { isPublished: boolean; fixStatus?: string }) => {
+                field.isPublished = isPublished;
+                if (this.fix) {
+                  field.fixStatus = fixStatus;
+                }
+              });
+  
+              if (this.fix && isPublished) {
+                this.log($t(auditFixMsg.ENTRY_MANDATORY_FIELD_FIX,{uid, locale:code}),'error')
+                entry.publish_details = [];
+              }
+            } else {
+              delete this.missingMandatoryFields[uid]
+            }
```

Comment:
@cs-raj , [Minor] at multiple place you're using this.missingMandatoryFields[uid]. Instead of using directly  this.missingMandatoryFields[uid] you can store it.
const fields = this.missingMandatoryFields[uid];
const hasFields = fields && entry.publish_details.length;

Reply:
added